src/App.jsx

```diff
@@ -14,45 +16,94 @@ export const goodsFromServer = [
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_OPTIONS = {
+  NONE: '',
+  ALPHABET: 'alphabet',
+  LENGTH: 'length',
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
+const GoodList = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li key={good} data-cy="Good">{good}</li>))}
+  </ul>
 );
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_OPTIONS.ALPHABET:
+          return good1.localeCompare(good2);
+        case SORT_OPTIONS.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SORT_OPTIONS.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods
+  = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', 'is-info',
+            { 'is-light': sortField !== SORT_OPTIONS.ALPHABET })}
+          onClick={() => {
+            setSortField(SORT_OPTIONS.ALPHABET);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-success',
+            { 'is-light': sortField !== SORT_OPTIONS.LENGTH })}
+          onClick={() => setSortField(SORT_OPTIONS.LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button', 'is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField(SORT_OPTIONS.NONE);
+              setIsReversed(false);
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)